tests/alembic_migrations/versions/v_7b139906ac46_add_not_null_constraint_on_meta_vars/test_smoke.py

```diff
@@ -0,0 +1,46 @@
+"""Smoke tests:
+- Upgrade adds not null constraint
+- Downgrade drops not null constraint
+"""
+
+# -*- coding: utf-8 -*-
+import logging
+import pytest
+from alembic import command
+from sqlalchemy import inspect
+
+logger = logging.getLogger("tests")
+
+@pytest.mark.usefixtures("new_db_left")
+def test_upgrade(prepared_schema_from_migrations_left, schema_name):
+    """test migration from 3d50ec832e47 to 7b139906ac46. """
```

Comment:
We also need a check of the data migration action. That would look like the following:
- Migrate to 3d50ec832e47
- Load `meta_vars` with rows containing nulls in the migration columns. You can probably do that with a few inserts just to `meta_vars`, rather than loading all kinds of stuff as the big behavioural tests do.
- Migrate to 7b139906ac46
- Check the effects of the data migration

Reply:
You can find an example of this kind of two-stage upgrade test in `tests/alembic_migrations/versions/v_e688e520d265_refactor_flags_association_tables`. The way the initial target revision for `test_upgrade` is set up is subtle (a synonym for confusing): It is overridden by the parametrization 

```
@pytest.mark.parametrize(
    "prepared_schema_from_migrations_left", ("bdc28573df56",), indirect=True
)
```

of that method. You can do the same, except using of course the prior revision to this one.

If you get too stuck or confused, don't hesitate to reach out for help.